pkg/security/security_profile/profile/profile.go

```diff
@@ -55,3 +77,156 @@ func NewSecurityProfile(selector cgroupModel.WorkloadSelector) *SecurityProfile
 		selector: selector,
 	}
 }
+
+type ProcessActivityNodeAndParent struct {
+	node   *dump.ProcessActivityNode
+	parent *ProcessActivityNodeAndParent
+}
+
+func NewProcessActivityNodeAndParent(node *dump.ProcessActivityNode, parent *ProcessActivityNodeAndParent) *ProcessActivityNodeAndParent {
+	return &ProcessActivityNodeAndParent{
+		node:   node,
+		parent: parent,
+	}
+}
+
+func ProcessActivityTreeWalk(processActivityTree []*dump.ProcessActivityNode,
+	walkFunc func(pNode *ProcessActivityNodeAndParent) bool) []*dump.ProcessActivityNode {
+	var result []*dump.ProcessActivityNode
+	if len(processActivityTree) <= 0 {
+		return result
+	}
+
+	var nodes []*ProcessActivityNodeAndParent
+	var node *ProcessActivityNodeAndParent
+	for _, n := range processActivityTree {
+		nodes = append(nodes, NewProcessActivityNodeAndParent(n, nil))
+	}
+	node = nodes[0]
+	nodes = nodes[1:]
+
+	for node != nil {
+		if walkFunc(node) {
+			result = append(result, node.node)
+		}
+
+		for _, child := range node.node.Children {
+			nodes = append(nodes, NewProcessActivityNodeAndParent(child, node))
+		}
+		if len(nodes) > 0 {
+			node = nodes[0]
+			nodes = nodes[1:]
+		} else {
+			node = nil
+		}
+	}
+	return result
+}
+
+func (p *SecurityProfile) findProfileProcessNodes(pc *model.ProcessContext) []*dump.ProcessActivityNode {
+	if pc == nil {
+		return []*dump.ProcessActivityNode{}
+	}
+
+	var parent *model.ProcessContext
+	if pc.Ancestor == nil {
```

Comment:
this if statement seems to be useless ? pc.GetNextAncestorBinary() returns already nil if the ancestor is nil ?